testcafe/tests/drive/drive-viewer-feature.js

```diff
@@ -0,0 +1,232 @@
+import { driveUser } from '../helpers/roles'
+import {
+  TESTCAFE_DRIVE_URL,
+  setDownloadPath,
+  getFilesWithExt,
+  checkLocalFile,
+  deleteLocalFile
+} from '../helpers/utils'
+let data = require('../helpers/data')
+import DrivePage from '../pages/drive-model'
+import ViewerPage from '../pages/drive-viewer-model'
+
+const drivePage = new DrivePage()
+const viewerPage = new ViewerPage()
+
+//************************
+//Tests when authentified
+//************************
+fixture`Drive : Viewer features : prepare data`.page`${TESTCAFE_DRIVE_URL}/`
+  .beforeEach(async t => {
+    console.group(`\n↳ ℹ️  Loggin & Initialization`)
+    await t.useRole(driveUser)
+    await drivePage.waitForLoading()
+    console.groupEnd()
+  })
+  .afterEach(async () => {
+    console.groupEnd()
+  })
+
+test('Drive : Create a $test_date_time folder in Drive', async () => {
+  console.group(`↳ ℹ️  Drive : Create a ${data.FOLDER_DATE_TIME} folder`)
+  await drivePage.addNewFolder(data.FOLDER_DATE_TIME)
+})
+
+test('Drive : Go to $test_date_time and upload 26 files', async () => {
+  console.group(
+    `↳ ℹ️  Drive : Go to ${data.FOLDER_DATE_TIME} and upload 26 files`
+  )
+  await drivePage.goToFolder(data.FOLDER_DATE_TIME)
+  await drivePage.uploadFiles(data.filesList)
+})
+
+//************************
+//Tests when authentified : wiht downloads
+//************************
+fixture`Drive : Viewer features (and Download)`
+  .page`${TESTCAFE_DRIVE_URL}/`.beforeEach(async t => {
+  console.group(
+    `\n↳ ℹ️  Loggin, Page Initialization & data.DOWNLOAD_PATH initialization`
+  )
+  await t.useRole(driveUser)
+  await drivePage.waitForLoading()
+  await drivePage.goToFolder(data.FOLDER_DATE_TIME)
+  //Files count needed for navigation tests
+  t.ctx.totalFilesCount = await drivePage.getContentRowCount(
+    `${data.FOLDER_DATE_TIME} - Before`
+  )
+  await setDownloadPath(data.DOWNLOAD_PATH)
+  console.groupEnd()
+})
+
+test('Viewer : checking commons features for all files (expect PDF)', async t => {
+  // put all files names , execpt pdf in an array for testing commons features in viewer
+  t.ctx.fileNameListNoPDF = await getFilesWithExt(
+    data.FILE_FROM_ZIP_PATH,
+    data.pdfFilesExt,
+    true
+  )
+  for (let i = 0; i < t.ctx.fileNameListNoPDF.length; i++) {
+    console.group(
+      `\n↳ ℹ️  Viewer : checking commons features for ${
+        t.ctx.fileNameListNoPDF[i]
+      }`
+    )
+    await viewerPage.checkCommonsViewerControlsAndDownload(
+      data.FOLDER_DATE_TIME,
+      t.ctx.fileNameListNoPDF[i]
+    )
+    console.groupEnd()
+  }
+}).after(async t => {
+  for (let i = 0; i < t.ctx.fileNameListNoPDF.length; i++) {
+    console.log(
+      `↳ ℹ️  Viewer (Commons) - Checking downloaded files for ${
+        t.ctx.fileNameListNoPDF[i]
+      }`
+    )
+    await checkLocalFile(`${data.DOWNLOAD_PATH}/${t.ctx.fileNameListNoPDF[i]}`)
```

Comment:
I disagree : as I use a loop this is only ONE big test checking all files (except pdf) for navigation, and download (I mean checking the button exist and that you can click on it). 
And when a test fail, testcafe stop the test, perform '.after`, and start the next test.

This way if for one file, the viewer page is OK (navigation, download button, etc...) but the file wasn't download, the test will keep running and check other files, and will only failed on `after`. So we know that the page is fine but this particular file has failed. 
Also using `after`or `afterEach`for checking file is an easy way to avoid timeout problem for downloaded  files as it will perform the whole test before checking on disk for file existance